src/java.base/share/classes/java/io/File.java

```diff
@@ -1986,6 +1987,18 @@ private TempDirectory() { }
         // temporary directory location
         private static final File tmpdir = new File(
                 GetPropertyAction.privilegedGetProperty("java.io.tmpdir"));
+
+        // print out the error message for java.io.tmpdir setting
+        static {
```

Comment:
This will only run when java.io.File.createTempFile is used. So it work won't work java.nio.files.Files.createTempXXX. Have you looked into testing the value of java.io.tmpdir at startup when it is specified on the command line?